persistent/Database/Persist/Quasi/Internal.hs

```diff
@@ -321,71 +321,86 @@ fixForeignKeysAll unEnts = map fixForeignKeys unEnts
     -- the names of the referenced columns
     fixForeignKey :: EntityDef -> UnboundForeignDef -> ForeignDef
     fixForeignKey ent (UnboundForeignDef foreignFieldTexts parentFieldTexts fdef) =
```

Comment:
this whole function should get trashed

we can't pass in the pre-existing entities here meaningfully so we shouldn't even be trying to get this right